### PR TITLE
feat:add fields for TTFB 

### PR DESCRIPTION
**Pull request by jiuker**
add field for TTFB

### CODE REVIEW

**Review by klauspost — COMMENTED** [review]
What is up with all the removed dependencies?

File: perf-site-replication.go [hunk 1]
```diff
@@ -63,3 +67,38 @@ func (adm *AdminClient) SiteReplicationPerf(ctx context.Context, duration time.D
...
+func (s *SiteNetPerfReader) Start() {
+	buf := make([]byte, 4*1024-1)
+	rand.Read(buf)
+	s.buf = buf
+	s.startTime = time.Now()
+}
+
+func (s *SiteNetPerfReader) Read(p []byte) (n int, err error) {
+	s.readCounter++
+	if s.readCounter == 1 {
```
Comment (harshavardhana) [inline perf-site-replication.go:89]: use switch case instead of double `if` @jiuker

File: perf-site-replication.go [hunk 2]
```diff
@@ -63,3 +67,38 @@ func (adm *AdminClient) SiteReplicationPerf(ctx context.Context, duration time.D
...
+		n = copy(p, []byte{1})
+	} else if s.readCounter == 2 {
+		s.TimeToFirstByte = time.Now().Sub(s.startTime)
```
Comment (harshavardhana) [inline perf-site-replication.go:92]:
```suggestion
		s.TimeToFirstByte = time.Since(s.startTime)
```

File: perf-site-replication.go [hunk 3]
```diff
@@ -63,3 +67,38 @@ func (adm *AdminClient) SiteReplicationPerf(ctx context.Context, duration time.D
...
+		n = copy(p, s.buf)
+	} else {
+		return 0, io.EOF
```
Comment (harshavardhana) [inline perf-site-replication.go:95]: You do not need to only send the first byte via this reader instead it must be the first Read() itself how much was received by the other end. 

Look at the server-side implementation of how a response recorder is used to calculate time to the first byte.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
don't use only `go mod tidy` - please use `go mod tidy -compat=1.19`

**Comment by jiuker** [comment]
@harshavardhana PTAL

**Pull request closed without merging**